### PR TITLE
[Go] Set Default Values for Required Variables where only one value is possible

### PR DESCRIPTION
**Pull request by prmedipa**
Fix [19107](https://github.com/OpenAPITools/openapi-generator/issues/19107)

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
Go tests (https://app.circleci.com/pipelines/github/OpenAPITools/openapi-generator/30653/workflows/271edded-3274-444b-b3b5-3a67e02e8c73/jobs/99661) failed. can you please take a look?

File: modules/openapi-generator/src/main/resources/go/model_simple.mustache [hunk 1]
```diff
@@ -356,20 +377,42 @@ func (o *{{{classname}}}) UnmarshalJSON(data []byte) (err error) {
 {{/requiredVars}}
 	}
 
+	defaultValueFuncMap := map[string]func() interface{} {
+{{#requiredVars}}
+{{#useDefaultValuesForRequiredVars}}
```
Comment (vvb) [inline modules/openapi-generator/src/main/resources/go/model_simple.mustache:382]: The existing generated code should not have any diff if the `useDefaultValueForRequiredVars` is false. From that perspective, should this be wrapping all the changed code?

Comment (prmedipa) [reply modules/openapi-generator/src/main/resources/go/model_simple.mustache:381]: I have wrapped whole newly added code under useDefaultValueForRequiredVars. Could you please check.

Comment (prmedipa) [reply modules/openapi-generator/src/main/resources/go/model_simple.mustache:381]: @wing328  @vvb  all test cases passed. Could you please review.

**Pull request closed without merging**